### PR TITLE
Drag file upload in Object Browser

### PR DESCRIPTION
**Pull request by jinapurapu**
**What does it do:** Files dropped into the Object Browser are uploaded to the server.
**How does it look:**

Screenshot from 2021-12-17 15-15-27
 Screenshot from 2021-12-17 15-15-38

#1053

### CODE REVIEW

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
 import React, { Fragment, useEffect, useRef, useState } from "react";
 import { connect } from "react-redux";
+import {useDropzone} from 'react-dropzone'
```
Comment (prakashsvmx) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx:19]: `react-dropzone` - Please update/commit `package.json`

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx [hunk 2]
```diff
@@ -81,6 +83,8 @@ import withSuspense from "../../../../Common/Components/withSuspense";
 import { displayName } from "./utils";
 import { DownloadIcon, UploadFolderIcon } from "../../../../../../icons";
 
+
```
Comment (prakashsvmx) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx:86]: Please remove extra blank line.

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx [hunk 3]
```diff
@@ -570,19 +581,27 @@ const ListObjects = ({
...
+      e.target.files === undefined
     ) {
       return;
     }
     e.preventDefault();
+    var newFiles : File[] = [];
 
-    let files = e.target.files;
+    for (var i=0; i<e.target.files.length; i++) {
+    newFiles.push(e.target.files[i])
```
Comment (prakashsvmx) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx:597]: When we drag and drop a Folder with nested folders it only uploads the files. 
Should the folder structure be preserved ? ( Prefix )

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx [hunk 4]
```diff
@@ -1099,11 +1121,19 @@ const ListObjects = ({
...
+                    id="file-input"
+                    style={{ display: "none" }}
+                    ref={folderUpload}
+                  />
                 </SecureComponent>
                 <input
                   type="file"
                   multiple
-                  onChange={(e) => uploadObject(e)}
+                  onChange={(e) => handleUploadButton(e)}
```
Comment (prakashsvmx) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx:1136]:
```suggestion
                  onChange={handleUploadButton}
```
the arguments are same. so an extra function can be avoided

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx [hunk 5]
```diff
@@ -1075,7 +1097,7 @@ const ListObjects = ({
...
```
Comment (prakashsvmx) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx:1100]:
```suggestion
                  onChange={handleUploadButton}
```
the arguments are same. so an extra function can be avoided

**Review by prakashsvmx — DISMISSED** [review]

**Comment by prakashsvmx** [comment]
Please fix the `yarn start` warning 

`src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx
  Line 259:6:  React Hook React.useCallback has a missing dependency: 'uploadObject'. Either include it or remove the dependency array  react-hooks/exhaustive-deps`

File: portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx (hunk 3 above)
Comment (dvaldivia) [reply portal-ui/src/screens/Console/Buckets/ListBuckets/Objects/ListObjects/ListObjects.tsx:597]: this is a great point, perhaps we should add the folder name as a prefix to all the files

**Review by dvaldivia — CHANGES_REQUESTED** [review]
Screen Shot 2021-12-22 at 9 44 58 PM

There's some weird double slash, I think it can be fixed easily

**Review by dvaldivia — APPROVED** [review]

**Comment by prakashsvmx** [comment]
@jinapurapu  

- [x] the nested folder gets uploaded as expected with prefix.
- [ ] the `//` is displayed Please check this. 
image

**Review by bexsoft — CHANGES_REQUESTED** [review]
There is an issue with folders drag and drop, after I try to do it it seems like files get uploaded but application becomes non responsive and don't show any information about uploads in upload manager. If I stop and start the server again I am able to see the files.

Screen Shot 2021-12-30 at 12 43 54

**Review by bexsoft — APPROVED** [review]
Comments will be handled in another PR. LGTM then

**Pull request merged**